### PR TITLE
feat(components): [tree-v2] support `node-drop` event

### PR DESCRIPTION
**Pull request by MoConWu**
更新[Tree-v2]控件，支持node的drop事件，事件名称：node-drop

/label Components:tree-v2

- [x] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [x] Make sure you are merging your commits to `dev` branch.
- [x] Add some descriptions and refer to relative issues for your PR.

### CODE REVIEW

**Comment by MoConWu** [comment]
> 👋 @MoConWu, thank you for contributing element-plus.
> 
> * You can comment with `/label Components:[component_name]` to add a label for which component you are working on.
> * You may join our Discord for staying tuned.

/label Components:tree-v2

**Comment by MoConWu** [comment]
审查员审查通过一下

**Comment by warmthsea** [comment]
@MoConWu 

可以提供一下极简的 playground 示例么

**Comment by MoConWu** [comment]
> @MoConWu
> 
> 可以提供一下极简的 playground 示例么

[playground 在线预览](https://element-plus.run/#eyJBcHAudnVlIjoiPHNjcmlwdCBzZXR1cCBsYW5nPVwidHNcIj5cbmltcG9ydCB7IHJlZiB9IGZyb20gXCJ2dWVcIlxuaW1wb3J0IHR5cGUgeyBUcmVlTm9kZURhdGEgfSBmcm9tICdlbGVtZW50LXBsdXMvZXMvY29tcG9uZW50cy90cmVlLXYyL3NyYy90eXBlcydcblxuaW50ZXJmYWNlIFRyZWUge1xuICBpZDogc3RyaW5nXG4gIGxhYmVsOiBzdHJpbmdcbiAgY2hpbGRyZW4/OiBUcmVlW11cbn1cblxuY29uc3QgZ2V0S2V5ID0gKHByZWZpeDogc3RyaW5nLCBpZDogbnVtYmVyKSA9PiB7XG4gIHJldHVybiBgJHtwcmVmaXh9LSR7aWR9YFxufVxuY29uc3QgY3JlYXRlRGF0YSA9IChcbiAgbWF4RGVlcDogbnVtYmVyLFxuICBtYXhDaGlsZHJlbjogbnVtYmVyLFxuICBtaW5Ob2Rlc051bWJlcjogbnVtYmVyLFxuICBkZWVwID0gMSxcbiAga2V5ID0gJ25vZGUnXG4pOiBUcmVlW10gPT4ge1xuICBsZXQgaWQgPSAwXG4gIHJldHVybiBBcnJheS5mcm9tKHsgbGVuZ3RoOiBtaW5Ob2Rlc051bWJlciB9KVxuICAgIC5maWxsKGRlZXApXG4gICAgLm1hcCgoKSA9PiB7XG4gICAgICBjb25zdCBjaGlsZHJlbk51bWJlciA9XG4gICAgICAgIGRlZXAgPT09IG1heERlZXAgPyAwIDogTWF0aC5yb3VuZChNYXRoLnJhbmRvbSgpICogbWF4Q2hpbGRyZW4pXG4gICAgICBjb25zdCBub2RlS2V5ID0gZ2V0S2V5KGtleSwgKytpZClcbiAgICAgIHJldHVybiB7XG4gICAgICAgIGlkOiBub2RlS2V5LFxuICAgICAgICBsYWJlbDogbm9kZUtleSxcbiAgICAgICAgY2hpbGRyZW46IGNoaWxkcmVuTnVtYmVyXG4gICAgICAgICAgPyBjcmVhdGVEYXRhKG1heERlZXAsIG1heENoaWxkcmVuLCBjaGlsZHJlbk51bWJlciwgZGVlcCArIDEsIG5vZGVLZXkpXG4gICAgICAgICAgOiB1bmRlZmluZWQsXG4gICAgICB9XG4gICAgfSlcbn1cblxuY29uc3QgZGF0YSA9IGNyZWF0ZURhdGEoMywgNSwgNSlcbmNvbnN0IHByb3BzID0ge1xuICB2YWx1ZTogJ2lkJyxcbiAgbGFiZWw6ICdsYWJlbCcsXG4gIGNoaWxkcmVuOiAnY2hpbGRyZW4nLFxufVxuY29uc3QgaW5mbyA9IHJlZignJylcbmNvbnN0IG9uX25vZGVfZHJvcCA9IChub2RlX2RhdGE6IFRyZWVOb2RlRGF0YSwgbm9kZTogYW55LCBldnQ6IERyYWdFdmVudCkgPT4ge1xuICBpbmZvLnZhbHVlID0gYFxuICBUcmVlTm9kZURhdGE6ICR7bm9kZV9kYXRhLmlkfVxuICBUcmVlTm9kZTogJHtub2RlfVxuICBEcmFnRXZlbnQ6ICR7ZXZ0fWBcbn1cbjwvc2NyaXB0PlxuXG48dGVtcGxhdGU+XG4gIDxwPuivt+aLluWKqFZ1ZeWbvueJh+WIsFRyZWUtdjI8L3A+XG4gIDxpbWcgc3JjPVwiaHR0cHM6Ly9lbGVtZW50LXBsdXMucnVuL2Fzc2V0cy9sb2dvLWtBeGpQaWVXLnN2Z1wiIHN0eWxlPVwid2lkdGg6IDEwMHB4OyBoZWlnaHQ6IDEwMHB4XCIgLz5cbiAgPGVsLXRyZWUtdjJcbiAgICAgIHN0eWxlPVwid2lkdGg6IDUwMHB4XCJcbiAgICAgIDpkYXRhPVwiZGF0YVwiXG4gICAgICA6cHJvcHM9XCJwcm9wc1wiXG4gICAgICA6aGVpZ2h0PVwiMjA4XCJcbiAgICAgIEBub2RlLWRyb3A9XCJvbl9ub2RlX2Ryb3BcIlxuICAgIC8+XG4gICAgPHAgc3R5bGU9XCJ3aGl0ZS1zcGFjZTogcHJlLXdyYXA7XCI+e3sgaW5mbyB9fTwvcD5cbjwvdGVtcGxhdGU+XG4iLCJlbGVtZW50LXBsdXMuanMiOiJpbXBvcnQgRWxlbWVudFBsdXMgZnJvbSAnZWxlbWVudC1wbHVzJ1xuaW1wb3J0IHsgZ2V0Q3VycmVudEluc3RhbmNlIH0gZnJvbSAndnVlJ1xuXG5sZXQgaW5zdGFsbGVkID0gZmFsc2VcbmF3YWl0IGxvYWRTdHlsZSgpXG5cbmV4cG9ydCBmdW5jdGlvbiBzZXR1cEVsZW1lbnRQbHVzKCkge1xuICBpZiAoaW5zdGFsbGVkKSByZXR1cm5cbiAgY29uc3QgaW5zdGFuY2UgPSBnZXRDdXJyZW50SW5zdGFuY2UoKVxuICBpbnN0YW5jZS5hcHBDb250ZXh0LmFwcC51c2UoRWxlbWVudFBsdXMpXG4gIGluc3RhbGxlZCA9IHRydWVcbn1cblxuZXhwb3J0IGZ1bmN0aW9uIGxvYWRTdHlsZSgpIHtcbiAgY29uc3Qgc3R5bGVzID0gWydodHRwczovL3ByZXZpZXctMTY5MzUtZWxlbWVudC1wbHVzLnN1cmdlLnNoL2J1bmRsZS9pbmRleC5jc3MnLCAnaHR0cHM6Ly9wcmV2aWV3LTE2OTM1LWVsZW1lbnQtcGx1cy5zdXJnZS5zaC9idW5kbGUvaW5kZXguY3NzJ10ubWFwKChzdHlsZSkgPT4ge1xuICAgIHJldHVybiBuZXcgUHJvbWlzZSgocmVzb2x2ZSwgcmVqZWN0KSA9PiB7XG4gICAgICBjb25zdCBsaW5rID0gZG9jdW1lbnQuY3JlYXRlRWxlbWVudCgnbGluaycpXG4gICAgICBsaW5rLnJlbCA9ICdzdHlsZXNoZWV0J1xuICAgICAgbGluay5ocmVmID0gc3R5bGVcbiAgICAgIGxpbmsuYWRkRXZlbnRMaXN0ZW5lcignbG9hZCcsIHJlc29sdmUpXG4gICAgICBsaW5rLmFkZEV2ZW50TGlzdGVuZXIoJ2Vycm9yJywgcmVqZWN0KVxuICAgICAgZG9jdW1lbnQuYm9keS5hcHBlbmQobGluaylcbiAgICB9KVxuICB9KVxuICByZXR1cm4gUHJvbWlzZS5hbGxTZXR0bGVkKHN0eWxlcylcbn1cbiIsInRzY29uZmlnLmpzb24iOiJ7XG4gIFwiY29tcGlsZXJPcHRpb25zXCI6IHtcbiAgICBcInRhcmdldFwiOiBcIkVTTmV4dFwiLFxuICAgIFwianN4XCI6IFwicHJlc2VydmVcIixcbiAgICBcIm1vZHVsZVwiOiBcIkVTTmV4dFwiLFxuICAgIFwibW9kdWxlUmVzb2x1dGlvblwiOiBcIkJ1bmRsZXJcIixcbiAgICBcInR5cGVzXCI6IFtcImVsZW1lbnQtcGx1cy9nbG9iYWwuZC50c1wiXSxcbiAgICBcImFsbG93SW1wb3J0aW5nVHNFeHRlbnNpb25zXCI6IHRydWUsXG4gICAgXCJhbGxvd0pzXCI6IHRydWUsXG4gICAgXCJjaGVja0pzXCI6IHRydWVcbiAgfSxcbiAgXCJ2dWVDb21waWxlck9wdGlvbnNcIjoge1xuICAgIFwidGFyZ2V0XCI6IDMuM1xuICB9XG59XG4iLCJQbGF5Z3JvdW5kTWFpbi52dWUiOiI8c2NyaXB0IHNldHVwPlxuaW1wb3J0IEFwcCBmcm9tICcuL0FwcC52dWUnXG5pbXBvcnQgeyBzZXR1cEVsZW1lbnRQbHVzIH0gZnJvbSAnLi9lbGVtZW50LXBsdXMuanMnXG5zZXR1cEVsZW1lbnRQbHVzKClcbjwvc2NyaXB0PlxuXG48dGVtcGxhdGU+XG4gIDxBcHAgLz5cbjwvdGVtcGxhdGU+XG4iLCJpbXBvcnQtbWFwLmpzb24iOiJ7XG4gIFwiaW1wb3J0c1wiOiB7XG4gICAgXCJ2dWVcIjogXCJodHRwczovL2Zhc3RseS5qc2RlbGl2ci5uZXQvbnBtL0B2dWUvcnVudGltZS1kb21AbGF0ZXN0L2Rpc3QvcnVudGltZS1kb20uZXNtLWJyb3dzZXIuanNcIixcbiAgICBcIkB2dWUvc2hhcmVkXCI6IFwiaHR0cHM6Ly9mYXN0bHkuanNkZWxpdnIubmV0L25wbS9AdnVlL3NoYXJlZEBsYXRlc3QvZGlzdC9zaGFyZWQuZXNtLWJ1bmRsZXIuanNcIixcbiAgICBcImVsZW1lbnQtcGx1c1wiOiBcImh0dHBzOi8vcHJldmlldy0xNjkzNS1lbGVtZW50LXBsdXMuc3VyZ2Uuc2gvYnVuZGxlL2luZGV4LmZ1bGwubWluLm1qc1wiLFxuICAgIFwiZWxlbWVudC1wbHVzL1wiOiBcInVuc3VwcG9ydGVkXCIsXG4gICAgXCJAZWxlbWVudC1wbHVzL2ljb25zLXZ1ZVwiOiBcImh0dHBzOi8vZmFzdGx5LmpzZGVsaXZyLm5ldC9ucG0vQGVsZW1lbnQtcGx1cy9pY29ucy12dWVAMi9kaXN0L2luZGV4Lm1pbi5qc1wiXG4gIH0sXG4gIFwic2NvcGVzXCI6IHt9XG59IiwiX28iOnsic2hvd0hpZGRlbiI6dHJ1ZSwic3R5bGVTb3VyY2UiOiJodHRwczovL3ByZXZpZXctMTY5MzUtZWxlbWVudC1wbHVzLnN1cmdlLnNoL2J1bmRsZS9pbmRleC5jc3MifX0=)

File: docs/en-US/component/tree-v2.md [hunk 1]
```diff
@@ -124,6 +124,7 @@ tree-v2/filter
 | Name             | Description                                          | Parameters                                                                                                                              |
 | ---------------- | ---------------------------------------------------- | --------------------------------------------------------------------------------------------------------------------------------------- |
 | node-click       | triggers when a node is clicked                      | `(data: TreeNodeData, node: TreeNode, e: MouseEvent)`                                                                                   |
+| node-drop        | triggers when drag someting and drop on a node       | `(data: TreeNodeData, node: TreeNode, e: DragEvent)`                                                                                    |
```
Comment (warmthsea) [inline docs/en-US/component/tree-v2.md:127]: Need to add the version number here.
^(2.8.2)

Comment (MoConWu) [reply docs/en-US/component/tree-v2.md]: OK

File: docs/en-US/component/tree-v2.md
Comment (warmthsea) [inline docs/en-US/component/tree-v2.md:1]: Need to be changed in the editor to ensure formatting alignment

Comment (MoConWu) [reply docs/en-US/component/tree-v2.md:1]: OK

File: docs/en-US/component/tree-v2.md (hunk 1 above)
Comment (warmthsea) [reply docs/en-US/component/tree-v2.md]: Sorry, I misread it, it should be 2.8.3

**Review by warmthsea — APPROVED** [review]
LGTM

**Comment by btea** [comment]
The code looks fine. 

However, I have a question: What is the purpose of adding this event? Do you want to insert elements into the node by dragging?

**Comment by MoConWu** [comment]
> The code looks fine.
> 
> However, I have a question: What is the purpose of adding this event? Do you want to insert elements into the node by dragging?

Yes, We can drag elements, files, text, anything we want to drag onto the node. These are user-defined, and once dragged in, an event will be triggered. Users can judge the dragging event themselves

**Review by btea — APPROVED** [review]
Thanks!

**Pull request merged**